scripts/legacy_expression_tool.md

```diff
@@ -0,0 +1,171 @@
+for each legacy expression found:
+
+- _id: The id of the object where the legacy expression is found
+- expression: The legacy expression found
+- type: The type of the object where the legacy expression is found (I.E: active.expression)
+- fiware_service: The fiware service of the object where the legacy expression is found
+- fiware_servicepath: The fiware service path of the object where the legacy expression is found
+
+### `legacy_expression_list.json`
+
+This file contains just a list of all the legacy expressions found in the database. It can be used to build the translation 
+file. The file looks like this:
+
+```json
+[
+    "${legacy-expression-1}",
+    "${legacy-expression-2}",
+    ...
+    "${legacy-expression-n}"
+]
+```
+
+### `objects_replaced.json`
+
+This file contains a list of objects where the legacy expressions have been replaced with all the information associated. This 
+file is useful to preview the changes expected to be done in the database before applying them.
+
+### `objects_backup.json`
+
+This file contains a list of objects containing legacy expression that have been backed up. This file is useful to preview the 
+changes expected to be done in the database before modifying it.
+
+## Command line arguments
+
+The script can be executed using the following command:
+
+```bash
+python legacy_expression_tool.py \
+    --host <mongodb-host> \
+    --port <mongodb-port> \
+    --database <mongodb-db> \
+    --collection <mongodb-collection> \
+    --translation <translation-file> 
+```
+
+The list of possible arguments that the scripts accepts are:
+
+| Argument | Description | Default value | Mandatory |
+| -------- | ----------- | ------------- | --------- |
+| `-H`, `--host` | The host of the MongoDB instance | `localhost` | No |
+| `-p`, `--port` | The port of the MongoDB instance | `27017` | No |
+| `-db`, `--database` | The database name to replace the expressions | NA | Yes |
+| `-c`, `--collection` | The collection name to replace the expressions | NA | Yes |
+| `-t`, `--translation` | The translation dictionary file to replace the expressions | `translation.json` | No |
+| `-D`, `--debug` | Enable debug mode | `False` | No |
+| `-cm`, `--commit` | Commit the changes to the database | `False` | No |
+| `-el`, `--expression-language` | What to do with the expression language field. Possibles values: `delete`, `ignore` or `jexl` | `ignore` | No |
+| `-stat`, `--statistics` | Print match statistics. Aggregation modes are the possible values: `service` and `subservice` | `service` | No |
+| `-S`, `--service` | The fiware service filter to replace the expressions | All subservices | No |
+| `-P`, `--service-path` | The fiware service path filter to replace the expressions | All subservices | No |
+| `-i`, `--deviceid` | The device id filter to replace the expressions | All devices | No |
```

Comment:
It's a matter of style, but instead of providing cryptic short CLI in a script which usage life is very limited (only during the migration) I'd suggest keep only the long options:

```suggestion
| Argument | Description | Default value | Mandatory |
| -------- | ----------- | ------------- | --------- |
| `--host` | The host of the MongoDB instance | `localhost` | No |
| `--port` | The port of the MongoDB instance | `27017` | No |
| `--database` | The database name to replace the expressions | NA | Yes |
| `--collection` | The collection name to replace the expressions | NA | Yes |
| `--translation` | The translation dictionary file to replace the expressions | `translation.json` | No |
| `--debug` | Enable debug mode | `False` | No |
| `--commit` | Commit the changes to the database | `False` | No |
| `--expression-language` | What to do with the expression language field. Possibles values: `delete`, `ignore` or `jexl` | `ignore` | No |
| `--statistics` | Print match statistics. Aggregation modes are the possible values: `service` and `subservice` | `service` | No |
| `--service` | The fiware service filter to replace the expressions | All subservices | No |
| `--service-path` | The fiware service path filter to replace the expressions | All subservices | No |
| `--deviceid` | The device id filter to replace the expressions | All devices | No |
```

Reply:
done in https://github.com/telefonicaid/iotagent-node-lib/pull/1381/commits/3e7c7b1e2b5300c32d6e5a3f13b9852b370de0c6